tensorflow/python/ops/array_ops.py

```diff
@@ -2300,13 +2300,20 @@ def _constant_if_small(value, shape, dtype, name):
   return None
 
 
-def _eager_mark_zeros_tensor(tensor):
-  if context.executing_eagerly():
-    setattr(tensor, "_is_zeros_tensor", True)
-  return tensor
+def _tag_zeros_tensor(fun):
+  """ Tags the result of function by setting _is_zeros_tensor attribute.
+
+  This is useful to compute Hessians of fused ops such as cross_entropy.
+  """
+  def wrapped(*args, **kwargs):
+    tensor = fun(*args, **kwargs)
+    tensor._is_zeros_tensor = True
+    return tensor
+  return wrapped
 
 
 @tf_export("zeros")
+@_tag_zeros_tensor
```

Comment:
I think as-is this will hide the tf.zeros docstring/args (and use the decorator's docstring instead). Can you use tf_decorator.make_decorator to make sure it gets copied over? https://github.com/tensorflow/tensorflow/blob/94adea511f3f8fcc64ab89569fce0ddbc0c1c82c/tensorflow/python/util/tf_decorator.py#L67

Example usage: https://github.com/tensorflow/tensorflow/blob/94adea511f3f8fcc64ab89569fce0ddbc0c1c82c/tensorflow/python/eager/function.py#L3184